qemu/tests/block_iscsi_4kdisk.py

```diff
@@ -0,0 +1,56 @@
+"""Test to install the Windows OS on the 4k disk."""
+import logging
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import utils_misc
+from virttest.iscsi import Iscsi
+from virttest.tests import unattended_install
+
+
+def run(test, params, env):
+    """
+    Test to install the guest OS on the 4k disk.
+    Steps:
+        1) Setup iSCSI server with fileio backend and enable 4k sector.
+        2) Discovery and login the above iSCSI target.
+        3) Mount the iscsi disk and create raw image on it.
+        4) Install a guest on the raw image as blk device and enable iothread .
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def _prepare():
+        cmd_prepare = params['cmd_prepare'].format(dev_name)
+        process.run(cmd_prepare, 600, shell=True)
+
+    def _cleanup():
+        if vm and vm.is_alive():
+            vm.destroy()
+        cmd_cleanup = params['cmd_cleanup'] % base_dir
+        process.run(cmd_cleanup, 600, shell=True)
+
+    try:
+        vm = None
+        params['image_size'] = params['emulated_image_size']
+        base_dir = data_dir.get_data_dir()
+        iscsi = Iscsi.create_iSCSI(params, base_dir)
+        iscsi.login()
+        dev_name = utils_misc.wait_for(lambda: iscsi.get_device_name(), 60)
+        if not dev_name:
+            test.error('Can not get the iSCSI device.')
+
+        logging.info('Prepare env on: %s', dev_name)
+        _prepare()
+        logging.info('Start to install ...')
+        vm = env.get_vm(params["main_vm"])
+        unattended_install.run(test, params, env)
+        logging.info('Install completed')
+        vm.destroy()
+        vm = None
+    finally:
+        _cleanup()
+        iscsi.cleanup()
```

Comment:
line 55 <->line 56 ? 
cleanup iscsi setup first, finally remove the emulated image

Reply:
iscsi can be None, it's better to check that before iscsi.cleanup

Reply:
iscsi must exist otherwise exception raised. so the check is not necessary.

Reply:
The cleanup umount first then can do iscsi clean.